### PR TITLE
[Messenger][Notifier] use more entropy with random_bytes()

### PR DESCRIPTION
**Pull request by xabbuh**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | https://github.com/symfony/symfony/pull/57856#discussion_r1694799065
| License       | MIT

### CODE REVIEW

**Comment by xabbuh** [comment]
based on [the PR feedback](https://github.com/symfony/symfony/pull/57856#discussion_r1694799065) I chose to pick Alexander's suggestion from #57588

File: src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php [hunk 1]
```diff
@@ -521,7 +522,7 @@ public function add(string $body, array $headers, int $delayInMs = 0): string
 
         try {
             if ($delayInMs > 0) { // the delay is <= 0 for queued messages
-                $id = bin2hex(random_bytes(4));
+                $id = (string) Uuid::v7();
```
Comment (derrabus) [inline src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php]: Would it make sense to inject a UUID factory into this connection?

Shall we use the base58 representation to get more compact identifiers?

**Review by derrabus — APPROVED** [review]
I think, switching to UUIDs for truly collision-free identifiers is a good call. Thank you for this PR.

File: src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php (hunk 1 above)
Comment (xabbuh) [reply src/Symfony/Component/Messenger/Bridge/Redis/Transport/Connection.php]: Using the base58 is a good idea. Using a factory IMO complicates things too much for the gain it brings.

**Review by derrabus — APPROVED** [review]

**Review by nicolas-grekas — APPROVED** [review]

**Comment by stof** [comment]
is it actually worth introducing a dependency to generate this (mostly internal) message identifier ? To me, using `bin2hex(random_bytes(...))` seems fine (still worth increasing the entropy by using 16 bytes instead of 4, matching the size of a UUID)

**Comment by derrabus** [comment]
We need random and collision-free identifiers, so we're pulling a dependency that does exactly that. This sounds absolutely reasonable to me.

**Comment by stof** [comment]
`bin2hex(random_bytes(16))` provides random identifiers without any dependency (we already depend on PHP), so I'm wondering whether we need that dependency.
A UUIDv7 provides additional features (time-sorting, standard UUID format), but I don't think those places need those additional benefits.

**Review by nicolas-grekas — COMMENTED** [review]
I think I agree with @stof, no need for the dep to dedup messages.
But I'd go with less entropy and a shorter hash (using 9 to generate tn trailing `=` with base64):
`base64_encode(random_bytes(9))`

File: src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php [hunk 2]
```diff
@@ -397,7 +397,7 @@ public static function provideIdPatterns(): \Generator
     {
         yield 'No delay' => ['/^THE_MESSAGE_ID$/', 0, 'xadd', 'THE_MESSAGE_ID'];
 
-        yield '100ms delay' => ['/^[a-f\d]+$/', 100, 'rawCommand', '1'];
+        yield '100ms delay' => ['/^[A-Z\d\/+]+={0,2}$/i', 100, 'rawCommand', '1'];
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php]: do we need to check for the the `=` at the end? 9 bytes never generates a `=`, does it?

Comment (xabbuh) [reply src/Symfony/Component/Messenger/Bridge/Redis/Tests/Transport/ConnectionTest.php]: it's indeed not needed

**Review by chalasr — APPROVED** [review]

**Review by derrabus — APPROVED** [review]

**Comment by nicolas-grekas** [comment]
Thank you @xabbuh.

**Pull request merged**